src/operators/reduce.rs

```diff
@@ -201,26 +201,26 @@ pub trait Count<G: Scope, K: Data, R: Semigroup> where G::Timestamp: Lattice+Ord
     /// This method allows `count` to produce collections whose difference
     /// type is something other than an `isize` integer, for example perhaps an
     /// `i32`.
-    fn count_core<R2: Abelian + From<i8>>(&self) -> Collection<G, (K, R), R2>;
+    fn count_core<R2: Ord + Abelian + From<i8> + 'static>(&self) -> Collection<G, (K, R), R2>;
 }
 
 impl<G: Scope, K: ExchangeData+Hashable, R: ExchangeData+Semigroup> Count<G, K, R> for Collection<G, K, R>
 where
     G::Timestamp: Lattice+Ord,
 {
-    fn count_core<R2: Abelian + From<i8>>(&self) -> Collection<G, (K, R), R2> {
+    fn count_core<R2: Ord + Abelian + From<i8> + 'static>(&self) -> Collection<G, (K, R), R2> {
         self.arrange_by_self_named("Arrange: Count")
             .count_core()
     }
 }
 
-impl<G, K: Data, T1, R: Semigroup> Count<G, K, R> for Arranged<G, T1>
+impl<G, K: Data, T1, R: Data+Semigroup+'static> Count<G, K, R> for Arranged<G, T1>
```

Comment:
Redundant static?

Reply:
Yes! Sorry, this was a weird one where because `R` shows up in the data it needed a `Data` constraint added back in.